### DISCUSSION TITLE
アナログ入力のセンサが繋がるようにできませんか？

### DISCUSSION THREAD

**Discussion by y-ohe** [algyan/XIAOGYAN#9] [Ideas]
ES基板の回路図を見ると、XIAO ESP32C3への入力には機能的にはロータリーエンコーダ入力、SW入力×2(SEL/ESC)とI2Cでの入出力になっていると思われます。
もうピンの空きがないのですが、どうにか工夫してAD入力のセンサを繋げられるようにできないでしょうか？
SPEAKER-ENABLEとENC-LEDを同じピンに割り当てて、空いた片方をAD入力にできないですかねー
AD入力が使えると、いろいろできることが増えると思われ、せっかくSPEAKERがつくので、MIC入力もできるようになると、もっと良いかなと思います。

**Comment by matsujirushi** [comment]
提案ありがとうございます！

具体的に、どんなものをADCに繋ぐことを想定していますか？

> マイクやりたい人は、Seeed Stduio XIAO nRF52840 Sense換装を案内しようと考えていました。😉

**Reply by y-ohe** [reply]
そうですね、マイク入力を一番で想定していました。
なるほど、Seeed Stduio XIAO nRF52840には、マイクセンサが登載されているんですね。
他には、電池駆動時のバッテリー残量(電圧)モニタ、水分センサあたりですね。

**Reply by tatsu1225** [reply]
確かにADはあるといいですね！と思いましたが，扱いやすそうな距離・光・温度などのセンサって，I2C or IOなGrove端子があればこと足りるな．って自己解決しました
マイクはサンプリングレートが入りそうなのでAD良さげですが，XIAO nRF52840で事足りそうですね...!!

**Reply by matsujirushi** [reply]
@y-ohe さん、 @tatsu1225 さん、ご返答ありがとうございます。
水分センサなどの値変化が遅いものは、I2Cで良いかなーと思いますがどうでしょう？（そういうセンサーが具体的にあるかは未確認ですが。） ADCだと校正や信号処理など面倒ですし。
電池駆動には、強い方が改造して作るのを想定しています。たしかにバッテリー残量でADC使いたいは”たしかに！”なので、改造するときにADCを確保しやすいように考慮しましょう。

**Reply by techan2007** [reply]
ADCが引き出してもらえると、うれしい、と思ったけれど、
ESP32 Devkitであったような、ADの直線性がない問題、ってのはXAIOでは解決されているのかしら？
XAIOはつかったことないし。
あと、バッテリー残量にADCつかうのってどうなんでしょうね？
うまくいくのかな？バッテリー駆動にするとVCC自体が減ってくるので
VREFがとれなくてうまくいかなさそうなんだけど。

**Reply by matsujirushi** [reply]
Seeed Studio XIAO nRF52840なら、それなりに測れそうな予感。（やってみたいとわからない）
![image](https://user-images.githubusercontent.com/8079404/209806136-85f72e84-e29f-454e-a11b-bef1b80d3176.png)

**Reply by y-ohe** [reply]
@techan2007 さん
> ADCが引き出してもらえると、うれしい、と思ったけれど、 ESP32 Devkitであったような、ADの直線性がない問題、ってのはXAIOでは解決されているのかしら？ XAIOはつかったことないし。 あと、バッテリー残量にADCつかうのってどうなんでしょうね？ うまくいくのかな？バッテリー駆動にするとVCC自体が減ってくるので VREFがとれなくてうまくいかなさそうなんだけど。

「バッテリー残量にADC」をというところですが、お手軽な乾電池を使うようなので、電池の種類やメーカ、ロットや単品毎に特性がばらつきすぎて最初からはそれほどまともな残量を得られるとは思っていません。
なので、ADCの直線性も気にしておらず、そこそこのバッテリー電圧が取得(計測)できてクラウドにデータをためて行ければ良いかと考えています(IoTデバイスなので、使用者の方でデータを収集し、ある程度データが揃ってくれば、特性も見えてきて、残量の方も分かってくるのかと)。
XIAOもバッテリー使うときは、直接ではなく、昇圧DCDCで3.3V を作るようなので、VREFもそれなりに安定すると思っています。

**Reply by techan2007** [reply]
なるほど〜
クラウド用にUSB接続あるなら問題ないし
昇降圧DCDCとかやるなら、いいのよね

直線性も補正かけちゃえばいい用途なら
ありよね

いずれにせよ、AD引き出しは賛成

**Reply by y-ohe** [reply]
@matsujirushi さん
> @y-ohe さん、 @tatsu1225 さん、ご返答ありがとうございます。 水分センサなどの値変化が遅いものは、I2Cで良いかなーと思いますがどうでしょう？（そういうセンサーが具体的にあるかは未確認ですが。） ADCだと校正や信号処理など面倒ですし。 電池駆動には、強い方が改造して作るのを想定しています。たしかにバッテリー残量でADC使いたいは”たしかに！”なので、改造するときにADCを確保しやすいように考慮しましょう。

たしかに、I2C接続のADCをXIAOの外部(基板内)に実装するのもありですね。
ただ、秋月のADCラインナップにはI2C接続できるADCはあまり多くないみたいです(だいたいがSPI接続のようです)。
それと、部品代アップが気になります。

**Reply by matsujirushi** [reply]
@y-ohe さん

> それと、部品代アップが気になります。

改造のコストは意識していないです。

ALGYAN 8周年イベントまでにバッテリーor乾電池駆動は諦めていて、将来バージョンで対応かなと思っています。
（初期バージョンをベテランの方々にいろいろといじって遊んでもらいたいところではあります。）https://github.com/algyan/algyan_8th#%E3%81%A9%E3%82%93%E3%81%AA%E3%83%87%E3%83%90%E3%82%A4%E3%82%B9%E3%81%AB%E3%81%97%E3%82%88%E3%81%86

**Comment by matsujirushi** [comment]
ES2基板で、XIAOのD3(A3)のパッドを用意しました→JP3。
XIAOピン不足のため、LEDと排他で使用できます。 ( https://github.com/algyan/algyan_8th/discussions/18#discussioncomment-4618862 )
